src/Spice86.Shared/Interfaces/ISerializableBreakpointsHolder.cs

```diff
@@ -0,0 +1,14 @@
+﻿namespace Spice86.Shared.Interfaces;
+
+using Spice86.Shared.Emulator.VM.Breakpoint.Serializable;
+
+/// <summary>
+/// This interface defines a method for generating a serializable object that encapsulates all breakpoints of the internal debugger
+/// </summary>
+public interface ISerializableBreakpointsHolder {
```

Comment:
A generic comment (and a bit personal taste): I see it as an anti pattern to use interfaces if there is only one implementation (and even more so if this is going to stay that way).

Reply:
It permits interface separation - no need to depend on an entire class when you only want a single method.

It can be expanded later to the GDB server so it can have its breakpoints serialized/restored too.